examples/greeter-extension/src/decorators.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-greeter-extension
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  ClassDecoratorFactory,
+  createViewGetter,
+  filterByTag,
+  inject,
+  MetadataInspector,
+} from '@loopback/context';
+
+export const EXTENSION_POINT_NAME = 'extensionPoint.name';
+
+/**
+ * Decorate a class as a named extension point. If the decoration is not
+ * present, the name of the class will be used.
+ *
+ * TODO: to be promoted to `@loopback/core` module.
+ * @param name Name of the extension point
+ */
+export function extensionPoint(name: string) {
+  return ClassDecoratorFactory.createDecorator(EXTENSION_POINT_NAME, {name});
+}
+
+/**
+ * Shortcut to inject extensions for the given extension point.
+ *
+ * TODO: to be promoted to `@loopback/core` module possibly as
+ * `@inject.extensions`.
+ *
+ * @param extensionPoint Name of the extension point. If not supplied, we use
+ * the name from `@extensionPoint` or the class name of the extension point.
+ */
+export function extensions(extensionPointName?: string) {
+  return inject('', {decorator: '@extensions'}, (ctx, injection, session?) => {
+    if (!session) return undefined;
```

Comment:
I might miss the implementation of session in other PRs, while wondering what does a `session` do?

Reply:
`session` is to track the dependency injection chain of bindings.